CondFormats/BeamSpotObjects/interface/SimBeamSpotObjects.h

```diff
@@ -42,31 +67,33 @@ class SimBeamSpotObjects {
   void setAlpha(double val) { fAlpha = val; }
   void setTimeOffset(double val) { fTimeOffset = val; }
 
-  /// get X position
+  /// get X, Y, Z position
   double x() const { return fX0; }
-  /// get Y position
   double y() const { return fY0; }
-  /// get Z position
   double z() const { return fZ0; }
-  /// get sigmaZ
+  /// get meanX, meanY, meanZ position
+  double meanX() const { return fMeanX; }
+  double meanY() const { return fMeanY; }
+  double meanZ() const { return fMeanZ; }
+  /// get sigmaX, sigmaY, sigmaZ
+  double sigmaX() const { return fSigmaX; }
```

Comment:
I think it would be nice if the CondFormat class is constructed with unphysical defaults for the sigmas (e.g. negative values) and IF the sigmas are not explicitly set by the client side (signifying it's a gaussian type smearing) we return here the expected widths from the beta function generator 1/sqrt(2) * sqrt(emittance * betastar)

Reply:
Ciao Marco, 

The only possible objection that comes to my mind is that when you inspect the content (via XML dump or via PayloadInspector), you can immediately recognize which variables are not used if they are all set to 0.0, e.g. in the BetaFunc smearing the sigmaX and sigmaY are not used, but with your suggestion they would be filled in any case, which might be a bit confusing.
But I must admit that there are also other variables that are used and (so far) always set to zero (e.g. `Phi`, `Alpha`...), so this objection isn't a very strong case, I just wrote it here for the records.

In any case I'm testing locally a commit which includes your suggestion and will push in few minutes.

Reply:
> via XML dump 

This will still display the default constructed value

Reply:
I have addressed this in a77360836b8cdd2ca93029505e73ffd883468feb, does that look like what you had in mind?
It produces something like this (for the BetaFunc generator case):
```
<cmsCondPayload class_id="0" tracking_level="0" version="0">
	<fX0>4.58531999999999967e-02</fX0>
	<fY0>-1.69659999999999984e-02</fY0>
	<fZ0>-7.49920000000000031e-02</fZ0>
	<fMeanX>0.00000000000000000e+00</fMeanX>
	<fMeanY>0.00000000000000000e+00</fMeanY>
	<fMeanZ>0.00000000000000000e+00</fMeanZ>
	<fSigmaX>7.67886710394182891e-04</fSigmaX>
	<fSigmaY>7.67886710394182891e-04</fSigmaY>
	<fSigmaZ>3.60000000000000009e+00</fSigmaZ>
	<fbetastar>3.00000000000000000e+01</fbetastar>
	<femittance>3.93099999999999993e-08</femittance>
	<fPhi>0.00000000000000000e+00</fPhi>
	<fAlpha>0.00000000000000000e+00</fAlpha>
	<fTimeOffset>0.00000000000000000e+00</fTimeOffset>
</cmsCondPayload>
```
and
![d60f7980-58aa-41bc-b046-17574151492f](https://github.com/cms-sw/cmssw/assets/7822641/241d1b51-3b64-4511-907e-dd45530bd276)

Reply:
Thanks,  but that's not what I had in mind. I was proposing to change the getter method, not the setter.

Reply:
> I was proposing to change the getter method, not the setter.

Oh ok I did not understand it. Ok I'll rework the commit then.

Just for my education: what is the advantage in updating the getter?

Reply:
That I don't have to remember the formula every time :)

Reply:
ahaha ok!
But isn't it the same to have the formula in the setter? probably I'm missing something...

Reply:
> But isn't it the same to have the formula in the setter

No, because as you pointed out that changes the XML dump, that is confusing. I just want s method that gives me physical width irrespective of the model

Reply:
Ok I should have addressed it in the last force push c7f6425ce1d0f4b3edad0ecee0b32cfe920361d1
Now the result as you pointed out is correctly different for xml dump and PayloadInspector:
```
<fX0>4.58531999999999967e-02</fX0>
<fY0>-1.69659999999999984e-02</fY0>
<fZ0>-7.49920000000000031e-02</fZ0>
<fMeanX>0.00000000000000000e+00</fMeanX>
<fMeanY>0.00000000000000000e+00</fMeanY>
<fMeanZ>0.00000000000000000e+00</fMeanZ>
<fSigmaX>-1.00000000000000000e+00</fSigmaX>
<fSigmaY>-1.00000000000000000e+00</fSigmaY>
<fSigmaZ>3.60000000000000009e+00</fSigmaZ>
<fbetastar>3.00000000000000000e+01</fbetastar>
<femittance>3.93099999999999993e-08</femittance>
<fPhi>0.00000000000000000e+00</fPhi>
<fAlpha>0.00000000000000000e+00</fAlpha>
<fTimeOffset>0.00000000000000000e+00</fTimeOffset>
```
and:
![a09fe6ca-558c-47fe-926a-369b9e9a984f](https://github.com/cms-sw/cmssw/assets/7822641/5c9af069-c9b2-4815-9b2b-e8286e62d025)

Reply:
Incidentally, now the line about sigma^{trans}_xy in the payload inspector could be dropped and the one about sigma_X and sigma_Y be converted in um.

Reply:
Addressed in 6d6fac25b8273ba37df2d82cf3f0f4c3a6c3e5dd (which is a squash of all the previous commits regarding the SimBeamSpot object).
I'll now move to the development of the new SimHLLHCBeamSpot CondFormat.